src/librustc_parse/parser/expr.rs

```diff
@@ -825,16 +825,12 @@ impl<'a> Parser<'a> {
             Ok(self.mk_expr(span, ExprKind::MethodCall(segment, args), AttrVec::new()))
         } else {
             // Field access `expr.f`
-            if let Some(args) = segment.args {
-                self.struct_span_err(
-                    args.span(),
-                    "field expressions may not have generic arguments",
-                )
-                .emit();
```

Comment:
I think your only recourse beyond doing the language change is to `.stash` the error instead and steal it in typeck. (And then add tests demonstrating that it still fails under `cfg(FALSE)`.)

Reply:
Please, no.
Just remove the second commit.
It's to much garbage for nearly zero benefit.

Reply:
(I'm just laying out the options; I don't have a particular opinion.)

Reply:
removed the commit

Reply:
removed the commit